src/Microsoft.AspNet.Mvc.Core/Infrastructure/ObjectResultExecutor.cs

```diff
@@ -159,6 +157,15 @@ public virtual Task ExecuteAsync(ActionContext context, ObjectResult result)
                 throw new ArgumentNullException(nameof(formatters));
             }
 
+            var responseContentType = formatterContext.HttpContext.Response.ContentType;
+            if (!string.IsNullOrEmpty(responseContentType))
+            {
+                contentTypes.Clear();
+                contentTypes.Add(MediaTypeHeaderValue.Parse(responseContentType));
+            }
+
+            ValidateContentTypes(contentTypes);
```

Comment:
This whole block should really be inside of `ExecuteAsync` (where you took this code from). If I'm overriding this method, I want to **use** what was passed in, not ignore it.